client/extensions/woocommerce/app/products/product-form-additional-details-card.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import { find, debounce } from 'lodash';
+import Gridicon from 'gridicons';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FoldableCard from 'components/foldable-card';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import FormTextInput from 'components/forms/form-text-input';
+import TokenField from 'components/token-field';
+
+class ProductFormAdditionalDetailsCard extends React.Component {
+
+	state = {
+		attributeNames: {},
+	};
+
+	static propTypes = {
+		product: PropTypes.object.isRequired,
+		editProduct: PropTypes.func.isRequired,
+		editProductAttribute: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.debouncedUpdateName = debounce( this.updateName, 300 );
+	}
+
+	getAttributes() {
+		const { product } = this.props;
+		return product.attributes && product.attributes.filter( attribute => ! attribute.variation ) || [];
+	}
+
+	getAttribute( { attributes }, attributeId ) {
+		return attributes && find( attributes, function( a ) {
+			return a.uid === attributeId;
+		} );
+	}
+
+	addAttribute = () => {
+		const { product, editProductAttribute } = this.props;
+		editProductAttribute( product, null, { name: '', options: [] } );
+	}
+
+	removeAttribute = ( e ) => {
+		const { product, editProduct } = this.props;
+		const attributes = [ ...this.getAttributes() ];
+		const attribute = this.getAttribute( product, e.currentTarget.id );
+		attributes.splice( attributes.indexOf( attribute ), 1 );
+		editProduct( product, { attributes } );
+	}
+
+	cardOpen = () => {
+		const attributes = this.getAttributes();
+		if ( ! attributes.length ) {
+			this.addAttribute();
```

Comment:
Doing this will make dirty-form detection a little tricker, but I can't think of a suggestion on how to avoid this without making rendering more complicated.